### PR TITLE
MB-6185 add admin endpoint list subscriptions

### PR DESCRIPTION
**Pull request by eamahanna**
## Description

This PR includes the following changes:
- updates admin.yaml to include webhookSubscriptions
- creates an IndexWebhookSubscriptionsHandler handler
- adds tests for the handler

## Setup
To test this endpoint you can execute `make server_test`

OR

Use postman to hit the endpoint
- Start the server `maker server_run`
- Start the client `make admin client_run`
- login to the admin app
- grab the admin_session_token cookie
- past this token value into the corresponding cookie in postman
- complete a GET request for `/webhook_subscriptions?filter=%7B%7D&order=true&page=1&perPage=25&sort=severity` or any variation of these filter variables
- you should see a response body that looks similar to the following:
```
[
    {
        "callbackUrl": "https://primelocal:9443/support/v1/webhook-notify",
        "createdAt": "2021-02-03T16:29:42.767Z",
        "eventKey": "PaymentRequest.Update",
        "id": "81a46d55-fd25-4ebc-ac13-bc311c7ce63e",
        "severity": 0,
        "status": "ACTIVE",
        "updatedAt": "2021-02-03T16:29:42.767Z"
    }
]
```

## Code Review Verification Steps

* [ ] If the change is risky, it has been tested in experimental before merging.
* [ ] Code follows the guidelines for [Logging](https://github.com/transcom/mymove/wiki/Backend-Programming-Guide#logging)
* [x] The requirements listed in [Querying the Database Safely](https://github.com/transcom/mymove/wiki/Backend-Programming-Guide#querying-the-database-safely) have been satisfied.
* Any new migrations/schema changes:
  * [ ] Follow our guidelines for zero-downtime deploys (see [Zero-Downtime Deploys](https://github.com/transcom/mymove/wiki/migrate-the-database#zero-downtime-migrations))
  * [ ] Have been communicated to #g-database
  * [ ] Secure migrations have been tested following the instructions in our [docs](https://github.com/transcom/mymove/wiki/migrate-the-database#secure-migrations)
* [ ] There are no aXe warnings for UI.
* [ ] This works in [Supported Browsers and their phone views](https://github.com/transcom/mymove/tree/master/docs/adr/0016-Browser-Support.md) (Chrome, Firefox, IE, Edge).
* [ ] Tested in the Experimental environment (for changes to containers, app startup, or connection to data stores)
* [ ] User facing changes have been reviewed by design.
* [ ] Request review from a member of a different team.
* [x] Have the Jira acceptance criteria been met for this change?

## References

* [Jira story](https://dp3.atlassian.net/browse/MB-6185) for this change
* [this article](tbd) explains more about the approach used.

### CODE REVIEW

File: swagger/admin.yaml [hunk 1]
```diff
@@ -863,6 +863,52 @@ definitions:
...
+        type: string
+        description: The URL to which the notifications for this subscription will be pushed to.
+      severity:
+        type: integer
+        minimum: 0
+      eventKey:
+        type: string
+        description: >-
+          A string used to represent which events this subscriber expects to be notified about.
+          Corresponds to the possible event_kek values in webhook_notifications.
```
Comment (abbyoung) [inline swagger/admin.yaml:905]: small typo

File: swagger/admin.yaml [hunk 2]
```diff
@@ -1641,3 +1687,43 @@ paths:
             $ref: '#/definitions/ValidationError'
         500:
           description: Server error
+  /webhook_subscriptions:
+    get:
+      summary: Lists webhook subscriptions
+      description: Returns a list of uwebhook subscriptions
```
Comment (abbyoung) [inline swagger/admin.yaml:1693]: small typo

File: pkg/handlers/adminapi/webhook_subscriptions_test.go [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+		response := handler.Handle(params)
+
+		suite.IsType(&webhooksubscriptionop.IndexWebhookSubscriptionsOK{}, response)
+		okResponse := response.(*webhooksubscriptionop.IndexWebhookSubscriptionsOK)
+		suite.Len(okResponse.Payload, 1)
+		suite.Equal(m.ID.String(), okResponse.Payload[0].ID.String())
+	})
+
+	suite.T().Run("404 - Move not found", func(t *testing.T) {
+		queryFilter := mocks.QueryFilter{}
```
Comment (abbyoung) [inline pkg/handlers/adminapi/webhook_subscriptions_test.go:46]: @shimonacarvalho has me adding more comments/info to the top of my tests, and I think it really helps future-us/other devs understand what the tests do, especially when there are mocks. Here's an example she shared with me:

```
// Under test: sendOneNotification function
		// Mocked:     Client
		// Set up:     We provide a FAILING webhook notification, and make the client fail twice to send
		//             and then succeed
		// Expected outcome:
		//             Client.Post called 3 times in total
		//             Notification would be updated as SENT```

File: pkg/handlers/adminapi/webhook_subscriptions.go [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+	services.ListFetcher
+	services.NewQueryFilter
+	services.NewPagination
+}
+
+// Handle retrieves a list of webhook subscriptions
+func (h IndexWebhookSubscriptionsHandler) Handle(params webhooksubscriptionop.IndexWebhookSubscriptionsParams) middleware.Responder {
+	logger := h.LoggerFromRequest(params.HTTPRequest)
+	// Here is where NewQueryFilter will be used to create Filters from the 'filter' query param
+	queryFilters := []services.QueryFilter{}
```
Comment (abbyoung) [inline pkg/handlers/adminapi/webhook_subscriptions.go:42]: Is this the part that is to comply with the react-admin pattern but not really used for this endpoint? if so, might be helpful to leave a comment so future devs understand its purpose?

**Review by abbyoung — APPROVED** [review]
looks and works great! i added a number of new subscriptions to my local db and tested sorting, pagination, order. left a few comments on some possible future-proofing via // comments ⚒️

File: pkg/handlers/adminapi/webhook_subscriptions.go (hunk 4 above)
Comment (eamahanna) [reply pkg/handlers/adminapi/webhook_subscriptions.go:42]: Chatted about this in slack :)

**Pull request merged**